exec/java-exec/src/main/java/org/apache/drill/exec/store/PluginBootstrapLoaderImpl.java

```diff
@@ -0,0 +1,298 @@
+
+    StoragePlugins updatedPlugins = updatedPlugins();
+    if (updatedPlugins != null) {
+      bootstrapPlugins.putAll(updatedPlugins);
+    }
+    return bootstrapPlugins;
+  }
+
+  @VisibleForTesting
+  protected StoragePlugins loadBootstrapPlugins() throws IOException {
+    // bootstrap load the config since no plugins are stored.
+    logger.info("No storage plugin instances configured in persistent store, loading bootstrap configuration.");
+    String storageBootstrapFileName = context.config().getString(ExecConstants.BOOTSTRAP_STORAGE_PLUGINS_FILE);
+    Set<URL> storageUrls = ClassPathScanner.forResource(storageBootstrapFileName, false);
+    // Format plugins does not work, see note below.
+//    String formatBootstrapFileName = context.config().getString(ExecConstants.BOOTSTRAP_FORMAT_PLUGINS_FILE);
+//    Set<URL> formatUrls = ClassPathScanner.forResource(formatBootstrapFileName, false);
+    if (storageUrls == null || storageUrls.isEmpty()) {
+      throw new IOException("Cannot find storage plugin boostrap file: " + storageBootstrapFileName);
+    }
+    logger.info("Loading the storage plugin configs from URLs {}.", storageUrls);
+    StoragePlugins bootstrapPlugins = new StoragePlugins();
+    Map<String, URL> pluginURLMap = new HashMap<>();
+    for (URL url : storageUrls) {
+      try {
+        loadStoragePlugins(url, bootstrapPlugins, pluginURLMap);
+      } catch (IOException e) {
+        throw new IOException("Failed to load bootstrap plugins from " + url.toString(), e );
+      }
+    }
+
+    // Note: this code doesn't work: format plugins are defined within DFS storage
+    // plugin configs; the can't be defined independently at present, alas.
+//    if (formatUrls != null && !formatUrls.isEmpty()) {
+//      logger.info("Loading the format plugin configs from URLs {}.", formatUrls);
+//      for (URL url : formatUrls) {
+//        loadFormatPlugins(url, bootstrapPlugins, pluginURLMap);
+//      }
+//    }
+    return bootstrapPlugins;
+  }
+
+  /**
+   * Get the new storage plugins from the
+   * {@link ConfigConstants#STORAGE_PLUGINS_OVERRIDE_CONF} file if it exists,
+   * null otherwise
+   */
+  @Override
+  public StoragePlugins updatedPlugins() {
+    String upgradeFileName = context.config().getString(ExecConstants.UPGRADE_STORAGE_PLUGINS_FILE);
+    Set<URL> urlSet = ClassPathScanner.forResource(upgradeFileName, false);
+    if (urlSet.isEmpty()) {
+      logger.trace("The {} file is absent. Proceed without updating of the storage plugins configs",
+          upgradeFileName);
+      return null;
+    }
+    if (urlSet.size() != 1) {
+      throw DrillRuntimeException.create(
+          "More than one %s file is placed in Drill's classpath: %s",
+          upgradeFileName, urlSet);
+    }
+    pluginsOverrideFileUrl = urlSet.iterator().next();
+    // Specialized form of the persistence mechanism
+    // to handle HOCON format in the override file
+    // TODO: not clear this needs to be in HOCON format.
```

Comment:
HOCON is also popular format. Now we allow both, for example, use on Web UI, you can export plugins in both formats.